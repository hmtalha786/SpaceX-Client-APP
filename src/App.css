body {
  background-image: url(https://free4kwallpapers.com/uploads/originals/2021/02/07/space-turksat-a-launch-wallpaper.jpg);
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  min-height: 100vh;
  color: #fff;
}

.status {
  color: "red";
}
