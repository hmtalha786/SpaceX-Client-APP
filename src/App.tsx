import React from "react";
import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
import "./App.css";
import Home from "./Components/Home/Home";
import LaunchPage from "./Components/Launch/LaunchPage";
import LaunchInfoPage from "./Components/LaunchInfo/LaunchInfoPage";
import MainNav from "./Components/MainNav/MainNav";
import RocketInfoPage from "./Components/RocketInfo/RocketInfoPage";
import RocketPage from "./Components/Rockets/RocketPage";
import ShipInfoPage from "./Components/ShipInfo/ShipInfoPage";
import ShipsPage from "./Components/Ships/ShipsPage";
import TopNav from "./Components/TopNav/TopNav";

function App() {
  return (
    <div className="App">
      <Router>
        <TopNav />
        <MainNav />
        <Routes>
          <Route path="/" element={<Home />} />
          <Route path="/launches" element={<LaunchPage />} ></Route>
          <Route path="/launches/:flight_number" element={<LaunchInfoPage />} /> 
          <Route path="/rockets" element={<RocketPage />} />
          <Route path="/rockets/:id" element={<RocketInfoPage />}  />
          <Route path="/ships" element={<ShipsPage />} />
          <Route path="/ships/:id" element={<ShipInfoPage />} />
        </Routes>
      </Router>
    </div>
  );
}

export default App;
