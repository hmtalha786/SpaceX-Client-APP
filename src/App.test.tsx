import React from 'react';
import {shallow} from 'enzyme';
import App from './App';
import { Route } from 'react-router-dom';

describe('renders the componenet' , () => {
    let container : any;

    beforeEach(() =>(container = shallow(<App />)))

    it('renders app component', () => {
        expect(container).toMatchSnapshot()
    })

    it('contains all the routes' , () =>{
        expect(container.find('div').length).toEqual(1);
    })

    it('contains route element' , () => {
        expect(container.containsMatchingElement(<Route />)).toEqual(true)
    })
    
})