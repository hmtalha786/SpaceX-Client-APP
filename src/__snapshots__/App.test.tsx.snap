// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the componenet renders app component 1`] = `
<div
  className="App"
>
  <BrowserRouter>
    <TopNav />
    <MainNav />
    <Routes>
      <Route
        element={<Home />}
        path="/"
      />
      <Route
        element={<LaunchPage />}
        path="/launches"
      />
      <Route
        element={<LaunchInfoPage />}
        path="/launches/:flight_number"
      />
      <Route
        element={<RocketPage />}
        path="/rockets"
      />
      <Route
        element={<RocketInfoPage />}
        path="/rockets/:id"
      />
      <Route
        element={<ShipsPage />}
        path="/ships"
      />
      <Route
        element={<ShipInfoPage />}
        path="/ships/:id"
      />
    </Routes>
  </BrowserRouter>
</div>
`;
